staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go

```diff
@@ -114,9 +114,26 @@ func ValidateFinalizerName(stringValue string, fldPath *field.Path) field.ErrorL
 		allErrs = append(allErrs, field.Invalid(fldPath, stringValue, msg))
 	}
 
+	if len(strings.Split(stringValue, "/")) == 1 {
+		if !IsStandardFinalizerName(stringValue) {
+			return append(allErrs, field.Invalid(fldPath, stringValue, "name is neither a standard finalizer name nor is it fully qualified"))
+		}
+	}
+
 	return allErrs
 }
 
+var standardFinalizers = sets.NewString(
+	metav1.FinalizerKubernetes,
+	metav1.FinalizerOrphanDependents,
+	metav1.FinalizerDeleteDependents,
+)
```

Comment:
standard or more like legacy?

Reply:
copy from:
https://github.com/kubernetes/kubernetes/blob/dc6d8c6b582dea09b355363af9612d1439dad09a/pkg/apis/core/helper/helpers.go#L292-L296

Reply:
my question is more of a meta question. If we forbid unqualified finalizer names everywhere, we probably should have followed the same rules with these "standard" finalizers. Any new finalizer added to the core system would likely be fully qualified. Hence, "legacy".

Reply:
yes, may be it should be named `legacyFinalizers` but not `standardFinalizers`.